src/main/java/aspects/SetBrowserValuesAspect.java

```diff
@@ -0,0 +1,32 @@
+package aspects;
+
+import beans.AuthenticatedRequestBean;
+import beans.SessionNavigationBean;
+import org.aspectj.lang.JoinPoint;
+import org.aspectj.lang.annotation.Aspect;
+import org.aspectj.lang.annotation.Before;
+import org.javarosa.core.model.utils.DateUtils;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.core.annotation.Order;
+import services.BrowserValuesProvider;
+
+/**
+ * Created by amstone326 on 1/8/18.
+ */
+@Aspect
+@Order(1)
+public class SetBrowserValuesAspect {
+
+    @Autowired
+    private BrowserValuesProvider browserValuesProvider;
+
+    @Before(value = "@annotation(org.springframework.web.bind.annotation.RequestMapping)")
+    public void setValues(JoinPoint joinPoint) throws Throwable {
+        Object beanArg = joinPoint.getArgs()[0];
+        if (beanArg instanceof AuthenticatedRequestBean) {
+            browserValuesProvider.setTimezoneOffset((AuthenticatedRequestBean)beanArg);
```

Comment:
If this condition isn't met looks like we'll still set the timezone provider but with the default timezone offset of `timezoneOffsetMillis`? I'd think if this condition isn't met we should log the event and not set the timezone provider?

Reply:
We need to set the provider every time because otherwise a request with a bean of some other type would end up using whatever `BrowserValuesProvider` had last been set as the provider in `DateUtils` by a previous request. But it's harmless to do so because the default value of `timezoneOffsetMillis in `BrowserValuesProvider` is -1, which is the condition under which we ignore the timezone offset and don't use it in `DateUtils`